src/test/java/com/puppycrawl/tools/checkstyle/JavaParserTest.java

```diff
@@ -205,9 +205,16 @@ public void testJava14TextBlocks() throws Exception {
     public void testNoStackOverflowOnDeepStringConcat() throws Exception {
         final File file =
                 new File(getPath("InputJavaParserNoStackOverflowOnDeepStringConcat.java"));
-        assertWithMessage("File parsing should not throw StackOverflowError.")
-                .that(JavaParser.parseFile(file, JavaParser.Options.WITH_COMMENTS))
-                .isNotNull();
+        final boolean isExecutableWithLimitedResources =
+                TestUtil.isExecutableWithLimitedResources(() -> {
+                    return JavaParser.parseFile(file, JavaParser.Options.WITH_COMMENTS);
+                });
+        final String message = "File parsing should not throw StackOverflowError"
+                + " with limited resources.";
+
+        assertWithMessage(message)
+                .that(isExecutableWithLimitedResources)
+                .isTrue();
```

Comment:
I made `TestUtil#isExecutableWithLimitedResources` return a boolean so that we can still use Truth to have fluent assertions in test methods, and avoid PMD's `JUnitTestsShouldIncludeAssert` rule.